aten/src/ATen/native/LinearAlgebraUtils.cpp

```diff
@@ -10,6 +10,14 @@ Tensor cloneBatchedColumnMajor(const Tensor& src) {
   desired_strides[last_dim] = src.size(second_to_last_dim);
   desired_strides[second_to_last_dim] = 1;
 
+  // Fast path for if the input already has correct strides
+  if (std::equal(desired_strides.begin(), desired_strides.end(), src.strides().begin())) {
+    auto result = src.transpose(-2, -1).clone();
+    result.transpose_(-2, -1);
+    return result;
```

Comment:
I thought this approach is always applicable, even if `src` does not have correct strides.

Reply:
good point!